l10n_ch_import_pain000/models/undo_payment_line.py

```diff
@@ -1,61 +1,119 @@
+                payment_order_obj.message_post(
+                    url + " has been removed because : " + data_supp[
+                        'add_tl_inf'])
+
+        else:
+            for account_move_line in account_move_lines:
+                # Create a link to the invoice that was removed
+                url = '<a href="web#id={}&view_type=form&model=account. ' \
+                      'invoice&action=241">{}</a>'. \
+                    format(account_move_line.invoice_id.id,
+                           account_move_line.invoice_id.move_name)
+                # Add a message to the invoice
+                account_move_line.invoice_id.message_post(
+                    "The invoice has been removed from the payment order,"
+                    " no reason given.")
+                # Add a message to the payment order
+                payment_order_obj.message_post(
+                    url + " has been removed no reason given.")
+
+    def _process_invoice_remove(self, account_payment_lines, payment_order_obj,
+                                data_supp):
+        """
+        This method unreconcile the given move lines and remove the payment.
+
+        :param account_payment_lines: The payment to remove
+        :param payment_order_obj: The payment order where to remove the payment
+        :param data_supp: additional information used in the message
+        :return: -
+        """
+
+        # Retrieve all account_move_line(s) associated with the previous
+        # account_payment_line(s)
+        account_move_lines = account_payment_lines.mapped(
+            'move_line_id')
+
+        full_reconcile_id = account_move_lines.mapped(
+            'full_reconcile_id').id
+
+        # Retrieve the counterpart of the previous move_line(s)
+        account_move_line_counterpart = self.env['account.move.line'].search([
+            ('full_reconcile_id', '=', full_reconcile_id),
+            ('id', 'not in', account_move_lines.mapped('id'))])
+
+        account_move_counterpart = account_move_line_counterpart.move_id
+
+        account_move_line_counterpart.remove_move_reconcile()
+
+        # unpost and delete the move from the journal.
+        account_move_counterpart.button_cancel()
+        account_move_counterpart.unlink()
+
+        # Delete the payment line in the payment order.
+        account_payment_lines.unlink()
+
+        # Search if there is something left in the payment order.
+        account_payment_line = self.env['account.payment.line'].search(
+            [('order_id', '=', payment_order_obj.id)])
+        if len(account_payment_line) == 0:
+            if payment_order_obj.state == 'uploaded':
+                payment_order_obj.action_done_cancel()
+
+        # Add the message to the invoice and to the payment order
+        self._post_message(data_supp, account_move_lines, payment_order_obj)
+
+    def undo_payment_line(self, payment_order_obj, bank_payment_line,
```

Comment:
Put the public method at first in your file and then the private methods. It will be more readable.